### ISSUE TITLE
What does an iteration mean?

### ISSUE THREAD

**Issue by kasri-mids** [LordMartian/deep_avsr#2]
I am unclear as to what an iteration means in the context of pretraining. As I understand, we run pretraining with increasing number of words (1, 2, 3 etc) where we pass forward the pretrained models from each run. I presume the pretrain needs to be run until convergence for each NUM_WORD (=1,2,etc). Is this what is referred to as an iteration? 

Thanks so much for your response and this repo.

**Comment by LordMartian** [comment]
An iteration is training the model for a particular PRETRAIN_NUM_WORDS value. In each iteration, the model is trained until the WER on the validation set stops decreasing. The number of words used in each iteration of curriculum learning is as follows: 1,2,3,5,7,9,13,17,21,29,37, i.e., 11 iterations in total.

You're welcome. 😀

**Comment by JackHenry1992** [comment]
@lordmartian Is this understanding correct

> we run pretraining with increasing number of words (1, 2, 3 etc) where we pass forward the pretrained models from each run

？

**Comment by lordmartian** [comment]
Thanks for your interest. Yes the understanding is correct.
